packages/create-workleap/src/bin.ts

```diff
@@ -1,111 +1,107 @@
-        note(`${outputDirectoryFromArgument} project setup`);
+        // bail if `force` is `false` or the user cancelled with Ctrl-C
+        if (force !== true) {
+            process.exit(1);
+        }
     }
+}
+
+const templateId = await p.select({
+    message: "What would you like to generate?",
+    initialValue: "host-application" as TemplateId,
+    options: [
+        {
+            value: "host-application",
+            label: "Host application"
+        },
+        {
+            value: "remote-module",
+            label: "Remote module"
+        },
+        {
+            value: "static-module",
+            label: "Static module"
+        }
+    ]
+});
 
-    return (
-        outputDirectoryFromArgument ??
-    (await text(
-        "Where should we create the project?",
-        DefaultOutputDirectory,
-        DefaultOutputDirectory
-    ))
-    );
-};
-
-const askForTemplateAsync = (): Promise<string> => {
-    const templates = [
-        { value: "host-application" },
-        { value: "remote-module" },
-        { value: "static-module" }
-    ];
-
-    return select<string>("Select the template to create", templates);
-};
-
-const askForScopeAsync = (template: string): Promise<string> => {
-    return text(
-        `What should be the ${template} scope?`,
-        "Press enter if no scope is needed."
-    );
-};
-
-const callFoundryAsync = async (
-    outputDirectory: string,
-    template: string,
-    scope: string
-): Promise<number> => {
-    const options = [template];
-
-    if (outputDirectory) {
-        options.push("-o", outputDirectory);
-    }
+if (p.isCancel(templateId)) { process.exit(1); }
 
-    if (scope) {
-        if (template === "host-application") {
-            options.push("--package-scope", scope);
-        } else {
-            options.push("--host-scope", scope);
-        }
-    }
+// Ask for other arguments
+let packageScope: string | undefined;
+let hostScope: string | undefined;
 
-    const childProcess = child_process.spawn(FoundryCmd, options, {
-        cwd: process.cwd(),
-        shell: true
-    });
+if (templateId === "host-application") {
+    packageScope = await p.text({
+        message: "What should be the package scope?",
+        placeholder: "  (hit Enter if no scope is needed)"
+    }) as string;
 
-    return new Promise(resolve => {
-        childProcess.stdout.on("data", (x: string): void => {
-            process.stdout.write(x.toString());
-        });
-        childProcess.stderr.on("data", (x: string): void => {
-            process.stderr.write(x.toString());
-            process.exit(1);
-        });
-        childProcess.on("exit", (code?: number): void => {
-            resolve(code ?? 0);
-        });
-    });
-};
+    if (p.isCancel(packageScope)) { process.exit(1); }
+} else {
+    hostScope = await p.text({
+        message: "What is the host application scope?",
+        placeholder: "  (hit Enter if no scope is needed)"
```

Comment:
The host application scope will always be required for remote module and static module templates